src/Configuration/Config.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -19,89 +17,70 @@ internal class JsonConfigurationFileParser
         private readonly Stack<string> _context = new Stack<string>();
         private string _currentPath;
 
-        private JsonTextReader _reader;
-
         public static IDictionary<string, string> Parse(Stream input)
             => new JsonConfigurationFileParser().ParseStream(input);
 
         private IDictionary<string, string> ParseStream(Stream input)
         {
             _data.Clear();
-            _reader = new JsonTextReader(new StreamReader(input));
-            _reader.DateParseHandling = DateParseHandling.None;
-
-            var jsonConfig = JObject.Load(_reader);
 
-            VisitJObject(jsonConfig);
+            using (var reader = new StreamReader(input))
```

Comment:
Can you pass the stream directly to the `JsonDocument.Parse` method instead?

Reply:
This causes some tests failures as it looks like the parser doesn't handle things like the byte order marker well, it results in parse errors with 0xEF

Reply:
cc @bartonjs, maybe this is something we should document.

Reply:
JsonDocument.Parse takes UTF-8.  Per The Unicode Consortium, there is no UTF-8 "BOM".  So that's actually an illegal start.

Reply:
> JsonDocument.Parse takes UTF-8. Per The Unicode Consortium, there is no UTF-8 "BOM". So that's actually an illegal start.

@bartonjs that's not accurate. There is a BOM `0xEF,0xBB,0xBF`, it's just not required. The parser should allow for it.
https://www.unicode.org/versions/Unicode5.0.0/ch02.pdf page 36
"Use of a BOM is neither required nor recommended for UTF-8, but may
be encountered in contexts where UTF-8 data is converted from other encoding forms that
use a BOM or where the BOM is used as a UTF-8 signature"

Reply:
Hm, clearly I've misremembered.  The current FAQ entry is pretty clear on "yes":

http://unicode.org/faq/utf_bom.html#bom5

> A: Yes, UTF-8 can contain a BOM.  ...

When I would have sworn that it said "this makes no sense, don't do it" (paraphrased).

@ahsonkhan Perhaps that means the right answer is that Utf8JsonReader should skip `0xEF, 0xBB, 0xBF` if that's the start of the data stream, since JsonDocument doesn't really have any understanding of what's going on (treat it as a special case of whitespace?)

Reply:
According to the JSON grammar (http://json.org/), only white space can be inserted between tokens which can be ignored and the BOM doesn't count as white space. However, the JSON RFC let's the parser decide what to do in the presence of a BOM. Do we **really** need our reader to ignore the BOM? If so, I am find with adding that logic (likely an extra if check in `ReadFirstToken`).

https://tools.ietf.org/html/rfc8259
> Implementations MUST NOT add a byte order mark (U+FEFF) to the
   beginning of a networked-transmitted JSON text.  In the interests of
   interoperability, implementations that parse JSON texts MAY ignore
   the presence of a byte order mark rather than treating it as an
   error.

Reply:
Discussing this further @GrabYourPitchforks, the layer that supports reading from a stream (or pipe) should strip out the BOM before passing the "text" data to the underlying reader. In this case, that should be JsonDocument The same would apply to the Deserializer. We shouldn't add the BOM stripping logic at the reader layer since we accept `ReadOnlySpan<byte>` which is meant to represent just the text data (without any metadata around it), and works as a stand-in for "utf8 string". Stream/etc. can contain this metadata.

cc @steveharter